src/nccl_ofi_net.c

```diff
@@ -2518,6 +2527,39 @@ ncclResult_t nccl_net_ofi_isend(void *sendComm, void* data, int size,
 		goto error;
 	}
 
+	nccl_ofi_comp = sComm->baseComm.ofi_comp;
+	if (OFI_UNLIKELY(nccl_ofi_comp == NULL)) {
+		ret = ncclSystemError;
+		NCCL_OFI_WARN("NCCL OFI component for dev %d is not initialised",
+			     sComm->dev);
+		goto error;
+	}
+
+	/*
+	 * In case, we are connecting to self, ensure that the request has
+	 * completed. If its completed, free the request. If not, progress the
+	 * function to process completions and return NULL request for send to
+	 * retry.
+	 */
+	if (OFI_UNLIKELY(sComm->connection_info && sComm->connection_info->connect_to_self == 1)) {
+		nccl_ofi_connection_info_t *conn_info = sComm->connection_info;
+		assert(conn_info->req != NULL);
+
+		if (conn_info->req->state == NCCL_OFI_REQ_COMPLETED) {
+			free_nccl_ofi_req(conn_info->req, false);
+			free(conn_info);
+		} else {
+			NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET,
+				       "Self-connect request: %p hasn't complete. Current State: %d",
```

Comment:
Nit: "completed."

Also, use state_str() instead of giving the user an integer for current state?